WebDriverAgentLib/Utilities/FBScreenshot.m

```diff
@@ -70,10 +71,18 @@ + (NSData *)takeWithQuality:(NSUInteger)quality
 {
   if ([self.class isNewScreenshotAPISupported]) {
     XCUIScreen *mainScreen = XCUIScreen.mainScreen;
+
+    // Note to fit to the old spec
+    CGRect newRect = rect;
+    newRect.origin.x = rect.origin.x * mainScreen.scale;
+    newRect.origin.y = rect.origin.y * mainScreen.scale;
+    newRect.size.height = rect.size.height * mainScreen.scale;
+    newRect.size.width = rect.size.width * mainScreen.scale;
+
     return [self.class takeWithScreenID:mainScreen.displayID
-                                  scale:mainScreen.scale
+                                  scale:SCREENSHOT_SCALE
                      compressionQuality:[self.class compressionQualityWithQuality:FBConfiguration.screenshotQuality]
-                                   rect:rect
+                                   rect:newRect
```

Comment:
This part could be confusing. In the corresponding method call we should decide whether we crop the screenshot after scaling or before it, so the passed `rect` coordinates should be already scaled or not. I would say it makes sense to cut the unscaled image and only after that scale the result

Reply:
This part is still not addressed. We should update the method documentation to know exactly whether the rect coordinates should be already scaled or not